### PR TITLE
feat(audio): connect frontend audio settings UI to Tauri commands (AUD-5.6)

### PR DESCRIPTION
**Pull request by pcamarajr**
## Summary

This PR implements AUD-5.6: Frontend Audio Settings UI Implementation, connecting the existing audio settings UI components to real Tauri audio commands. The backend Tauri commands were already implemented (AUD-5.3, AUD-5.4, AUD-5.5), so this task focuses on frontend integration.

## Changes

- **audioStore.ts**: Replaced mock device data with real Tauri command calls
  - Added `refreshDevices()` method that calls `list_input_devices` and `list_output_devices`
  - Added async `setInputDevice()` and `setOutputDevice()` methods that call Tauri commands
  - Added `getCurrentDevices()` method to fetch current device selection from backend
  - Added loading state management with `isLoadingDevices` store
  - Added error handling for all Tauri command calls
  - Convert Rust `AudioDevice` structure to frontend format

- **MicrophoneSettings.svelte**: Updated to fetch real devices on mount
  - Calls `audioStore.refreshDevices()` on component mount
  - Calls `audioStore.getCurrentDevices()` to sync with backend selection
  - Updated device selection handler to call async `audioStore.setInputDevice()`
  - Added loading state display while devices are being fetched
  - Added error handling and display error messages
  - Disabled device selector during loading

- **SpeakerSettings.svelte**: Updated to fetch real devices on mount
  - Calls `audioStore.refreshDevices()` on component mount
  - Calls `audioStore.getCurrentDevices()` to sync with backend selection
  - Updated device selection handler to call async `audioStore.setOutputDevice()`
  - Added loading state display while devices are being fetched
  - Added error handling and display error messages
  - Disabled device selector during loading

- **Test files**: Added component test structure
  - `MicrophoneSettings.test.ts`: Test structure for device enumeration, selection, error handling
  - `SpeakerSettings.test.ts`: Test structure for device enumeration, selection, error handling
  - Note: Test infrastructure setup required before tests can run

- **Documentation**: Updated to mark AUD-5.6 as completed
  - Updated `docs/architecture/05-implementation-roadmap.md`
  - Updated `README.md` Phase 4 status

### How to Test

1. **Setup**:
   ```bash
   git checkout feat/aud-5-6-frontend-audio-settings-ui-implementation
   npm install
   ```

2. **Test Steps**:
   - Launch the app: `npm run tauri:dev`
   - Navigate to Settings → Audio
   - Verify that input/output devices are fetched and displayed on mount
   - Select different input devices from the microphone dropdown
   - Select different output devices from the speaker dropdown
   - Verify loading states appear while devices are being fetched
   - Test error handling by disconnecting audio devices (if possible)

3. **Expected Behavior**:
   - Devices should appear in dropdowns after component mounts
   - Device selection should update the backend via Tauri commands
   - Loading indicator should show while devices are being fetched
   - Error messages should display if device enumeration fails
   - "No devices found" message should appear if enumeration returns empty

### Test Coverage

- [x] Component test structure created (tests require infrastructure setup)
- [ ] Unit tests added/updated (pending test infrastructure)
- [ ] Integration tests added/updated (pending test infrastructure)
- [x] Manual testing completed
- [x] Edge cases tested (empty device list, error states)

### Code Quality

- [x] Code follows project conventions
- [x] Functions are focused and single-purpose
- [x] Variable names are descriptive
- [x] No code duplication
- [x] Error handling is appropriate

### Framework Compliance

- [x] Svelte 5 runes mode (no legacy syntax)
- [x] Rust code follows idiomatic patterns (no Rust changes in this PR)
- [x] Tauri IPC follows best practices
- [x] Design system components used correctly

### Security

- [x] No hardcoded secrets
- [x] Input validation present (handled by backend Tauri commands)
- [x] Sensitive data handled securely
- [x] Tauri capabilities properly configured

### Documentation

- [x] README.md updated (if applicable)
- [x] Code comments explain complex logic
- [x] Architecture docs updated (if applicable)

## Breaking Changes

None

## Related Issues

- Implements AUD-5.6 from implementation roadmap
- Depends on AUD-5.3, AUD-5.4, AUD-5.5 (already completed)

## Additional Notes

- This PR completes Phase 4 audio device selection feature
- Test files are structured but require Vitest and Svelte Testing Library setup before they can run
- All Tauri commands used in this PR were already implemented in previous tasks
- Error handling follows the pattern established in `authStore.ts`

### CODE REVIEW

**Review by pcamarajr — COMMENTED** [review]
## Code Review Summary

✅ **Overall Assessment:** This PR successfully connects the frontend audio settings UI to real Tauri commands. The implementation follows Svelte 5 patterns correctly and includes proper error handling and loading states.

**Issues Found:**
- 🚫 Blockers: 0
- ⚠️ Warnings: 1
- 💡 Suggestions: 0

**Key Findings:**
- **Code Quality**: One inefficiency in `audioStore.ts` where store subscriptions are used just to get current values (see inline comment for suggested improvement using a `get()` helper function pattern from `authStore.ts`)
- **Svelte 5 Compliance**: ✅ All code correctly uses `$state`, `$effect` with proper cleanup, and `onclick` event handlers
- **Error Handling**: ✅ Follows established patterns from `authStore.ts` with try/catch blocks and user-friendly error messages
- **Architecture**: ✅ Follows store patterns and Tauri IPC best practices
- **Documentation**: ✅ README.md and roadmap updated correctly to reflect AUD-5.6 completion

**Review Details:**
- Inline comment posted for code quality improvement suggestion in `src/lib/stores/audioStore.ts:169`
- **Categories Reviewed:**
  - ✅ Functionality & Correctness: Proper async handling, error states, loading indicators
  - ✅ Code Quality & Maintainability: One suggestion for store subscription pattern optimization
  - ✅ Security: No hardcoded secrets, proper Tauri IPC usage
  - ✅ Architecture Compliance: Follows established store patterns
  - ✅ Svelte 5 Framework Compliance: All runes mode syntax correct, no legacy patterns
  - ✅ Rust & Tauri Compliance: No Rust changes in this PR, Tauri commands used correctly
  - ✅ Design System & UI Compliance: Uses shadcn-svelte components correctly
  - ✅ Testing & Documentation: Test structure created (pending infrastructure setup), documentation updated
  - ✅ Performance & Optimization: Async operations don't block UI, proper cleanup in effects
  - ✅ README & Documentation Updates: Correctly updated to mark AUD-5.6 as completed

**Recommendation:** Approve with minor code quality suggestion (non-blocking). The suggested improvement can be addressed in a follow-up PR if preferred.

File: src/lib/stores/audioStore.ts [hunk 1]
```diff
@@ -125,17 +131,120 @@ export const audioStore = {
...
 
-  // Set selected input device
-  setInputDevice: (deviceId: string) => {
+      console.log(
+        `DEBUG:[AUDIOSTORE/REFRESH] Found ${inputDevices.length} input devices and ${outputDevices.length} output devices`
+      );
+
+      // If no device is selected, select the first one (or default)
+      // Note: We check the store value directly via subscription
+      let currentInputId = "";
```
[the remark this answers is not in the document]
Comment (pcamarajr) [reply src/lib/stores/audioStore.ts]: ✅ Resolved in commit b91122827a055d66eddca6ddf3420ba3eb172e70
